src/plugins/workspace/public/components/workspace_creator/workspace_form.tsx

```diff
@@ -141,6 +150,38 @@ export const WorkspaceForm = ({
     [defaultVISTheme]
   );
 
+  const allFeatures = useMemo(
+    () =>
+      featureOrGroups.reduce<WorkspaceFeature[]>(
+        (previousData, currentData) => [
+          ...previousData,
+          ...(isWorkspaceFeatureGroup(currentData) ? currentData.features : [currentData]),
+        ],
+        []
+      ),
+    [featureOrGroups]
+  );
+
+  const featureDependencies = useMemo(
+    () =>
+      allFeatures.reduce<{ [key: string]: string[] }>(
```

Comment:
Can you leave a comment here? It's hard to understand the code. Or maybe extract it to a function? `function agoodname(allFeatures) {...}`